api/src/main/java/net/kyori/adventure/text/TranslatableComponent.java

```diff
@@ -27,8 +27,21 @@
 import org.checkerframework.checker.nullness.qual.NonNull;
 
 /**
- * A translatable component.
+ * A component that can display translated text.
  *
+ * <p>This component consists of:</p>
+ * <ul>
+ *   <li>key - a translation key used together with the viewer locale to fetch a translated string.</li>
+ *   <li>args(optional) - components that can be used as arguments in the translated string.
+ *   <p>(e.g "You picked up <b>{0}</b>." -&#62; "You picked up <b>Carrot</b>")</p></li>
+ * </ul>
+ *
+ * <p>Displaying this component through an {@link net.kyori.adventure.audience.Audience} will run it through the global translator by default,
+ * if there is any registered translations for the key it will be translated into the locale
+ * closest to the locale of the audience, optionally using the arguments on any placeholders in the string.</p>
```

Comment:
This should also mention that if no translation is found in the server-side store, the translation will be passed through to the client which will perform translation using any keys defined in an active resource pack.

Reply:
I was going to remove this since it felt like an implementation detail, do you think the default platform behavior should be described here still?

Reply:
The way to think about this is that any documentation can also be considered part of the API. If the intention of the API is to do this, and thus something _all_ implementations must follow, then it should be documented. If this is implementation dependent, then it shouldn't.

From @zml2008's comment, I would expect that what they said is intended as an API contract, in which case you'd include it.